Correctly handle /usr/gnu/bin/stty sane on Solaris.

This problem has been fixed upstream.

See:

http://debbugs.gnu.org/cgi/bugreport.cgi?bug=23866
http://git.sv.gnu.org/gitweb/?p=coreutils.git;a=commitdiff;h=v8.25-41-gdfae782

--- src/stty.c.orig	2016-06-28 17:48:25.580754994 -0700
+++ src/stty.c	2016-06-29 06:19:49.159950703 -0700
@@ -118,9 +118,9 @@
 # define CSWTCH _POSIX_VDISABLE
 #endif
 
-/* SunOS 5.3 loses (^Z doesn't work) if 'swtch' is the same as 'susp'.
+/* SunOS >= 5.3 loses (^Z doesn't work) if 'swtch' is the same as 'susp'.
    So the default is to disable 'swtch.'  */
-#if defined __sparc__ && defined __svr4__
+#if defined __sun
 # undef CSWTCH
 # define CSWTCH _POSIX_VDISABLE
 #endif
